src/toxcore/dht/packet/cookie.rs

```diff
@@ -145,22 +145,20 @@ impl EncryptedCookie {
     - fails to decrypt
     - fails to parse `Cookie`
     */
-    pub fn get_payload(&self, symmetric_key: &secretbox::Key) -> Result<Cookie, Error> {
+    pub fn get_payload(&self, symmetric_key: &secretbox::Key) -> Result<Cookie, GetPayloadError> {
         let decrypted = secretbox::open(&self.payload, &self.nonce, symmetric_key)
             .map_err(|()| {
                 debug!("Decrypting Cookie failed!");
-                Error::new(ErrorKind::Other, "Cookie decrypt error.")
+                GetPayloadError::DecryptError { packet: "Cookie".to_string() }
             })?;
         match Cookie::from_bytes(&decrypted) {
-            IResult::Incomplete(e) => {
-                debug!(target: "Dht", "Cookie return deserialize error: {:?}", e);
-                Err(Error::new(ErrorKind::Other,
-                    format!("Cookie return deserialize error: {:?}", e)))
+            IResult::Incomplete(needed) => {
+                debug!(target: "Dht", "Cookie return deserialize error: {:?}", needed);
+                Err(GetPayloadError::IncompletePayload { packet: "Cookie".to_string(), needed, payload: self.payload.to_vec() })
             },
-            IResult::Error(e) => {
-                debug!(target: "Dht", "Cookie return deserialize error: {:?}", e);
-                Err(Error::new(ErrorKind::Other,
-                    format!("Cookie return deserialize error: {:?}", e)))
+            IResult::Error(error) => {
+                debug!(target: "Dht", "Cookie return deserialize error: {:?}", error);
+                Err(GetPayloadError::DeserializeError { packet: "Cookie".to_string(), error, payload: self.payload.to_vec() })
```

Comment:
Cookie is not a packet. Isn't it strange that `DeserializeError` contains `packet` field here? Do we need this field at all?